pkg/krmtotf/fetchlivestate.go

```diff
@@ -53,15 +53,12 @@ func FetchLiveState(ctx context.Context, resource *Resource, provider *tfschema.
 
 }
 
-// Special handling for KMSCryptoKey that still lives after its parent KMSKeyRing is deleted.
+// SkipOrphanedCheck Special handling for KMSCryptoKey that still lives after its parent KMSKeyRing is deleted.
 // We can import the tf state directly from itself instead of sourcing for its parent.
 // More info in b/279485255#comment14
-func shouldGetImportIDFromSelfForDelete(resource *Resource) bool {
-	return resource.Kind == "KMSCryptoKey"
-}
-
-func ShouldResolveParentForDelete(resource *Resource) bool {
-	return !shouldGetImportIDFromSelfForDelete(resource) || hasEmptySelfLink(resource)
+func SkipOrphanedCheck(resource *Resource) bool {
+	allowlist := []string{"KMSCryptoKey"}
+	return contains(allowlist, resource.Kind) || !hasEmptySelfLink(resource)
```

Comment:
IIUC we have special handling for orphaned resource because we are not able to resolve its parent and then get the unique id for deletion. If the selfLink is non-empty, then we do not need to resolve parent and can simply use selfLink as the unique id and proceed the deletion(see fetchLiveStateFromId). Similar to Justin's comment.

However, that would cause redundant API call, since we assume if a resource is orphaned, then it's no longer exist in the underlying GCP.

Reply:
Ah makes sense!